### PR TITLE
LDMSD configuration counter

### PR DESCRIPTION
**Pull request by nick-enoent**
Add configuration counter that increments when a ldmsd successfully completes a configuration request

Config counter only increments for requests that modify the current configuration of the ldmsd

### CODE REVIEW

**Comment by nick-enoent** [comment]
@narategithub @nichamon Hey guys, any thoughts on this PR as implemented? It creates a "configuration counter" that increments with each successful configuration request that modifies a ldmsd's current configuration. i.e. configuration requests such as prdcr_stats will not increment the counter. 

In order to do this, I utilized a switch statement with all of the commands that do _not_ modify the ldmsd's current configuration. I could see this being mildly irritating to maintain in the future when/if we introduce new status commands, as they will have to be manually added to this switch statement.

An alternative I considered was creating an array of all "status" config requests in a centralized location.

File: ldms/src/ldmsd/ldmsd.c [hunk 1]
```diff
@@ -436,6 +437,16 @@ const char *ldmsd_loglevel_to_str(enum ldmsd_loglevel level)
 	return "LDMSD_LNONE";
 }
 
+void ldmsd_inc_cfg_cntr()
+{
+	__sync_fetch_and_add(&cfg_cntr, 1);
```
Comment (narategithub) [inline ldms/src/ldmsd/ldmsd.c:442]: From https://gcc.gnu.org/onlinedocs/gcc/_005f_005fsync-Builtins.html:

> These functions are implemented in terms of the ‘__atomic’ builtins (see __atomic Builtins). They should not be used for new code which should use the ‘__atomic’ builtins instead.

So, I suggest that you use `__atomic_fetch_add(&cfg_cntr, 1, __ATOMIC_SEQ_CST)` instead.

**Review by narategithub — COMMENTED** [review]
Just a minor suggestion about `__sync_fetch_and_add()` (please see in-line comment). Everything else looks fine to me.

File: ldms/src/ldmsd/ldmsd.c (hunk 1 above)
Comment (nick-enoent) [reply ldms/src/ldmsd/ldmsd.c:442]: Thanks @narategithub, I'll update that. I actually ran into the __atomic builtins when I was implementing this, but figured I'd try to stay consistent with the existing code.

**Comment by nichamon** [comment]
> @narategithub @nichamon Hey guys, any thoughts on this PR as implemented? It creates a "configuration counter" that increments with each successful configuration request that modifies a ldmsd's current configuration. i.e. configuration requests such as prdcr_stats will not increment the counter.
> 
> In order to do this, I utilized a switch statement with all of the commands that do _not_ modify the ldmsd's current configuration. I could see this being mildly irritating to maintain in the future when/if we introduce new status commands, as they will have to be manually added to this switch statement.
> 
> An alternative I considered was creating an array of all "status" config requests in a centralized location.

Two other methods are 1) to use the flag in the request_handler_entry struct https://github.com/ovis-hpc/ovis/blob/1e83ade3aa723c269507dc0d0c269951e08fa4f9/ldms/src/ldmsd/ldmsd_request.c#L186 and 2) to add another field in the request_handler_entry struct to differentiate between status and non-status request commands. For the prior, you could pick a bit and check the bit instead of the switch statement. Either of the methods will force us to specify whether the request command is a status or not in the request_handler[] array. https://github.com/ovis-hpc/ovis/blob/1e83ade3aa723c269507dc0d0c269951e08fa4f9/ldms/src/ldmsd/ldmsd_request.c#L299

**Pull request merged**